instances.go

```diff
@@ -249,3 +347,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
 
 	return nil
 }
+
+type portsMetadata struct {
+	CollectionType string `json:"collectionType"`
+	NumberOfPorts  int    `json:"collectionTotal"`
+}
+
+type portResponse struct {
+	Metadata portsMetadata `json:"metadata"`
+	Ports    []privatePort `json:"resources"`
+}
+type privatePort struct {
+	PrivatePort int `json:"private_endpoint_port,omitempty"`
```

Comment:
The standard for our json is camel case. Is this used for encoding or decoding data? If encoding, it should be `private_endpoint_port` -> `privatePort` or `privateEndpointPort` (if all other port references are endpointPort).